### PR TITLE
[misc] Error handle and TLS

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -473,7 +472,7 @@ def check_version():
     try:
         payload = json.dumps(payload)
         payload = payload.encode()
-        req = request.Request('http://metadata.taichi.graphics/check_version',
+        req = request.Request('https://metadata.taichi.graphics/check_version',
```
Comment (ailzhang) [inline python/taichi/lang/__init__.py:475]: In the future we might want to change this without changing source code ;)

**Review by ailzhang — APPROVED** [review]
LGTM!

File: python/taichi/lang/__init__.py (hunk 1 above)
Comment (Leonz5288) [reply python/taichi/lang/__init__.py:475]: Good point. Fetching from vault might be a solution in the future.

**Pull request merged**